src/main/java/org/openstreetmap/atlas/checks/validation/linear/edges/SinkIslandCheck.java

```diff
@@ -78,7 +82,9 @@ public boolean validCheckForObject(final AtlasObject object)
                 && ((Edge) object).highwayTag()
                         .isMoreImportantThanOrEqualTo(this.minimumHighwayType)
                 && !(this.isServiceRoad((Edge) object)
-                        && this.isWithinAreasWithExcludedAmenityTags((Edge) object));
+                        && this.isWithinAreasWithExcludedAmenityTags((Edge) object))
+                && !(Validators.isOfType(object, HighwayTag.class, HighwayTag.SERVICE)
```

Comment:
This seems somewhat redundant with the `this.isServiceRoad()` call above.

Reply:
In case of roads that are within airports or that ends in buildings, we are considering all service roads and not just the ones that have service tag along with highway=service. I could refactor the code to use the validations in `isServiceRoad()` in line if the method call is confusing.

Reply:
Why do the uses of isServiceRoad require there to be a service tag?

Reply:
I think that was part of the previous enhancements were the use cases required the roads to be having both highway=service and service=* tags.

Reply:
Ok, it could be worth reviewing those cases to see if we can combine things. Assuming this is the way it needs to be, though, it would be worth added a comment or two about this.